### ISSUE TITLE
Dependency Issues

### ISSUE THREAD

**Issue by carnalim** [sksamuel/elastic4s#667]
Whenever I try and run a simple JAR application I keep getting this error:

```
Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/logging/log4j/Logger
	at org.elasticsearch.common.logging.Loggers.getLogger(Loggers.java:105)
	at org.elasticsearch.common.logging.Loggers.getLogger(Loggers.java:72)
	at org.elasticsearch.common.component.AbstractComponent.<init>(AbstractComponent.java:37)
	at org.elasticsearch.plugins.PluginsService.<init>(PluginsService.java:110)
	at org.elasticsearch.client.transport.TransportClient.newPluginService(TransportClient.java:81)
	at org.elasticsearch.client.transport.TransportClient.buildTemplate(TransportClient.java:106)
	at org.elasticsearch.client.transport.TransportClient.<init>(TransportClient.java:228)
	at org.elasticsearch.transport.client.PreBuiltTransportClient.<init>(PreBuiltTransportClient.java:69)
	at org.elasticsearch.transport.client.PreBuiltTransportClient.<init>(PreBuiltTransportClient.java:65)
	at com.sksamuel.elastic4s.ElasticClient$.transport(ElasticClient.scala:101)
	at com.sksamuel.elastic4s.ElasticClient$.transport(ElasticClient.scala:67)
	at com.mattc.elastic.main$.main(main.scala:15)
	at com.mattc.elastic.main.main(main.scala)
Caused by: java.lang.ClassNotFoundException: org.apache.logging.log4j.Logger
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 13 more
```

**Comment by sksamuel** [comment]
Are you using the embedded mode?

**Comment by SharkyLV** [comment]
Same here. Not using embedded mode.

Got it working by importing many random libraries it asked for - log4j, netty, apache http, 100x lucene...

**Comment by sksamuel** [comment]
It's elasticsearch that has the deps on them. Shouldn't need them for the
transport client though you'd think.

**Comment by sksamuel** [comment]
Ok I know the fix for this, should be able to release 5.0.1 by tomorrow.

**Comment by fancellu** [comment]
Aha, I see exactly the same using 5.0.0

connecting to standalone, i.e.

     val uri = ElasticsearchClientUri("elasticsearch://localhost:9300")

**Comment by sksamuel** [comment]
So I know what the issue is, but not sure how to fix it. Elasticsearch transport client has compile time dependencies in the pom, but it seems SBT ignores that.

**Comment by sksamuel** [comment]
Should be fixed in 5.0.1

**Comment by michaelqiu94** [comment]
I'm running a JAR application using 5.0.1 and am getting the following:

Exception in thread "main" java.lang.NoSuchMethodError: io.netty.buffer.CompositeByteBuf.addComponents(ZLjava/lang/Iterable;)Lio/netty/buffer/CompositeByteBuf;
	at org.elasticsearch.transport.netty4.Netty4Utils.toByteBuf(Netty4Utils.java:75)
	at org.elasticsearch.transport.netty4.Netty4Transport.sendMessage(Netty4Transport.java:449)
	at org.elasticsearch.transport.netty4.Netty4Transport.sendMessage(Netty4Transport.java:91)
	at org.elasticsearch.transport.TcpTransport.internalSendMessage(TcpTransport.java:947)
	at org.elasticsearch.transport.TcpTransport.sendRequest(TcpTransport.java:929)
	at org.elasticsearch.transport.TransportService.sendRequestInternal(TransportService.java:493)
	at org.elasticsearch.transport.TransportService.sendRequest(TransportService.java:459)
	at org.elasticsearch.transport.TransportService.submitRequest(TransportService.java:445)
	at org.elasticsearch.client.transport.TransportClientNodesService$SimpleNodeSampler.doSample(TransportClientNodesService.java:388)
	at org.elasticsearch.client.transport.TransportClientNodesService$NodeSampler.sample(TransportClientNodesService.java:324)
	at org.elasticsearch.client.transport.TransportClientNodesService.addTransportAddresses(TransportClientNodesService.java:191)
	at org.elasticsearch.client.transport.TransportClient.addTransportAddress(TransportClient.java:282)
	at com.sksamuel.elastic4s.ElasticClient$$anonfun$transport$2.apply(ElasticClient.scala:103)
	at com.sksamuel.elastic4s.ElasticClient$$anonfun$transport$2.apply(ElasticClient.scala:102)
	at scala.collection.TraversableLike$WithFilter$$anonfun$foreach$1.apply(TraversableLike.scala:733)
	at scala.collection.immutable.List.foreach(List.scala:381)
	at scala.collection.TraversableLike$WithFilter.foreach(TraversableLike.scala:732)
	at com.sksamuel.elastic4s.ElasticClient$.transport(ElasticClient.scala:102)
	at com.sksamuel.elastic4s.ElasticClient$.transport(ElasticClient.scala:67)
